CHANGELOG.md

```diff
@@ -50,6 +50,10 @@ or use UDL with types from more than one crate.
 
 - Added the `FfiType::MutReference` variant.
 
+- `Type::Custom` is no longer turned in to a `Type::External` for the bindings. Binding authors
+  must check the type is local themselves before deciding to treat it as a local or external type.
+  We expect `Type::External` will be removed soon and all types will be treated in this way.
```

Comment:
An alternative to this would be to lean in to `Type::External` rather than away from it:
 - Remove the `module_path` from each type variant -- types are assumed to be from the current module
 - `Type::External` contains a `module_path` field and an `inner` field that can wraps another variant.

I like your approach better though.  The code changes seem simpler.  Also, maybe we can handle the internal/external distinction at another level.  For example, in my bindings pipeline PR, we could add a `TypeDefinition::External` variant.  I'm not sure exactly why, but that feels better to me.

Reply:
I agree with this and think having the bindings themselves make the distinction is a good one - it's much like what this patch already does - `Type::Custom` isn't represented by `Type::External`, but there *is* still external type templates for used for it. You will probably not be too surprised to hear I've got another PR on top of this to kill `Type::External` entirely, but that still uses the same approach - the external templates still exist.